stl/inc/flat_map

```diff
@@ -705,188 +610,209 @@ public:
+    template <class _OtherKey>
+    pair<iterator, iterator> equal_range(const _OtherKey& _Key_val)
         requires _Is_transparent_v<key_compare>
     {
-        return _Equal_range(_X);
+        return _Equal_range(_Key_val);
     }
 
-    pair<const_iterator, const_iterator> equal_range(const key_type& _X) const {
-        return _Equal_range(_X);
+    pair<const_iterator, const_iterator> equal_range(const key_type& _Key_val) const {
+        return _Equal_range(_Key_val);
     }
 
-    template <class _K>
-    pair<const_iterator, const_iterator> equal_range(const _K& _X) const
+    template <class _OtherKey>
+    pair<const_iterator, const_iterator> equal_range(const _OtherKey& _Key_val) const
         requires _Is_transparent_v<key_compare>
     {
-        return _Equal_range(_X);
+        return _Equal_range(_Key_val);
     }
 
-    friend bool operator==(const _Derived& _X, const _Derived& _Y) {
-        return _RANGES equal(_X._Data.keys, _Y._Data.keys) && _RANGES equal(_X._Data.values, _Y._Data.values);
+    friend bool operator==(const _Derived& _Left, const _Derived& _Right) {
+        return _RANGES equal(_Left._Data.keys, _Right._Data.keys)
+            && _RANGES equal(_Left._Data.values, _Right._Data.values);
     }
 
-    friend auto operator<=>(const _Derived& _X, const _Derived& _Y) {
+    friend auto operator<=>(const _Derived& _Left, const _Derived& _Right) {
         return _STD lexicographical_compare_three_way(
-            _X.cbegin(), _X.cend(), _Y.cbegin(), _Y.cend(), _Synth_three_way{});
+            _Left.cbegin(), _Left.cend(), _Right.cbegin(), _Right.cend(), _Synth_three_way{});
     }
 
-    friend void swap(_Derived& _X, _Derived& _Y) noexcept {
-        _X.swap(_Y);
+    friend void swap(_Derived& _Left, _Derived& _Right) noexcept {
+        _Left.swap(_Right);
     }
 
 
-private:
+protected:
     key_compare _Key_compare;
     containers _Data;
 
-    template <class _K1, class _K2>
-        requires (same_as<remove_cvref_t<_K1 &&>, key_type> && same_as<remove_cvref_t<_K2 &&>, key_type>)
-              || (constructible_from<key_type, _K1 &&> && constructible_from<key_type, _K2 &&>
+    template <class _Predicate>
+    size_type _Erase_if(_Predicate _Pred) {
+        _Clear_flat_map_scope_guard<_Flat_map_base> _Guard{this};
+
+        auto _View           = _View_to_mutate();
+        auto _Mut_first      = _View.begin();
+        auto _Mut_last       = _View.end();
+        const auto _Old_size = size();
+
+        _STD _Seek_wrapped(_Mut_first, _RANGES remove_if(_View, _Pred).begin());
+
+        erase(const_iterator{_Mut_first._Key_it, _Mut_first._Mapped_it},
+            const_iterator{_Mut_last._Key_it, _Mut_last._Mapped_it});
+
+        _Guard._Clearable = nullptr;
+        return _Old_size - size();
+    }
+
+private:
+    template <class _KeyTy1, class _KeyTy2>
+        requires (same_as<remove_cvref_t<_KeyTy1 &&>, key_type> && same_as<remove_cvref_t<_KeyTy2 &&>, key_type>)

```

Comment:
`remove_cvref_t<_KeyTy1 &&>` and `remove_cvref_t<_KeyTy2 &&>` don't need to say `&&` as it will be immediately removed.

Occurs below, e.g. `remove_cvref_t<_OtherKey &&>`.